### PR TITLE
Initial implementation of the toggling features process

### PR DESCRIPTION
**Pull request by t13ka**
- rework of the features branch in the settings_data.json file
- added new constants of the feature states

### CODE REVIEW

File: client-app/src/shims-tsx.d.ts [hunk 1]
```diff
@@ -18,5 +18,24 @@ declare global {
...
+    ProductsFilteringFeature: boolean;
+    ProductDetailsBrowsingFeature: boolean;
+    ProductPriceBrowsingFeature: boolean;
+    ProductRecommendationFeature: boolean;
+    OrderDraftFeature: boolean;
+    WishListFeature: boolean;
+    SubmitOrderFeature: boolean;
+    ProductInventoryBrowsingFeature: boolean;
+    ManageShipmentDetailsFeature: boolean;
+    ManagePaymentDetailsFeature: boolean;
```
Comment (asvishnyakov) [inline client-app/src/shims-tsx.d.ts:39]: May we move it to single object?

File: client-app/src/core/constants.ts [hunk 2]
```diff
@@ -34,3 +34,24 @@ export const invoicesGridFields = window.THEME_SETTINGS.invoices_grid_fields as
...
+export const productsFilteringFeature = window.ProductsFilteringFeature;
+export const productDetailsBrowsingFeature = window.ProductDetailsBrowsingFeature;
+export const productPriceBrowsingFeature = window.ProductPriceBrowsingFeature;
+export const productRecommendationFeature = window.ProductRecommendationFeature;
+export const orderDraftFeature = window.OrderDraftFeature;
+export const wishListFeature = window.WishListFeature;
+export const submitOrderFeature = window.SubmitOrderFeature;
+export const productInventoryBrowsingFeature = window.ProductInventoryBrowsingFeature;
+export const manageShipmentDetailsFeature = window.ManageShipmentDetailsFeature;
+export const managePaymentDetailsFeature = window.ManagePaymentDetailsFeature;
```
Comment (asvishnyakov) [inline client-app/src/core/constants.ts:56]: May we move it to single object?

File: layout/theme.liquid [hunk 3]
```diff
@@ -31,6 +31,26 @@
...
+      window.ProductDetailsBrowsingFeature = {{ 'ProductDetailsBrowsing' | is_feature_active }};
+      window.ProductPriceBrowsingFeature = {{ 'ProductPriceBrowsing' | is_feature_active }};
+      window.ProductRecommendationFeature = {{ 'ProductRecommendation' | is_feature_active }};
+      window.OrderDraftFeature = {{ 'OrderDraft' | is_feature_active }};
+      window.WishListFeature = {{ 'WishList' | is_feature_active }};
+      window.SubmitOrderFeature = {{ 'SubmitOrder' | is_feature_active }};
+      window.ProductInventoryBrowsingFeature = {{ 'ProductInventoryBrowsing' | is_feature_active }};
+      window.ManageShipmentDetailsFeature = {{ 'ManageShipmentDetails' | is_feature_active }};
+      window.ManagePaymentDetailsFeature = {{ 'ManagePaymentDetails' | is_feature_active }};
+
```
Comment (asvishnyakov) [inline layout/theme.liquid:53]: And same here about single object

File: config/settings_data.json [hunk 4]
```diff
@@ -538,139 +538,158 @@
...
+    {
+      "Name": "OrdersBrowsing",
       "IsActive": true,
-      "Behavior": {
-        "Requires": ["PaymentBrowsing"],
-        "Replaces": []
-      }
+      "Requires": [],
+      "Replaces": [],
+      "Replaced": [],
```
Comment (trueboroda) [inline config/settings_data.json:547]: Why are these two parameters necessary? Is this not redundant?

File: client-app/src/shims-tsx.d.ts (hunk 1 above)
Comment (t13ka) [reply client-app/src/shims-tsx.d.ts:43]: we may. it's done

File: client-app/src/core/constants.ts (hunk 2 above)
Comment (t13ka) [reply client-app/src/core/constants.ts]: we may. it's done

File: layout/theme.liquid (hunk 3 above)
Comment (t13ka) [reply layout/theme.liquid]: it's done

File: config/settings_data.json (hunk 4 above)
Comment (t13ka) [reply config/settings_data.json]: Agree. The replaced property was removed

**Pull request merged**